pycbc/inference/sampler/dynesty.py

```diff
@@ -98,9 +112,35 @@ def __init__(self, model, nlive, nprocesses=1,
                                                   prior_call, self.ndim,
                                                   nlive=self.nlive,
                                                   pool=pool, **kwargs)
+        self._sampler.pool = pool
+        self._sampler.M = pool.map
 
     def run(self):
-        self._sampler.run_nested(**self.run_kwds)
+        diff_niter = 1
+        if self.run_with_checkpoint is True:
+            if self.checkpoint_file:
+                try:
+                    self.resume_from_checkpoint()
+                    self.niter = self._sampler.results.niter
+                    logging.info('Successfully read from the checkpoint file')
+                except KeyError:
```

Comment:
Is a KeyError the only possible error?

Reply:
At least in the testing it was the only error I encountered in various possible scenarios.

Reply:
If the key is there but is corrupt, this won't be the error. This is one of the examples, where a blanket except would be correct.